### PR TITLE
feat(bigquery): expose customer managed encryption key for ML models

### PR DESCRIPTION
**Pull request by HemangChothani**
issue [9251]

### CODE REVIEW

File: bigquery/tests/unit/model/test_model.py [hunk 1]
```diff
@@ -283,6 +299,18 @@ def test_replace_labels(object_under_test):
     assert object_under_test.labels == {}
 
 
+def test_set_encryption_configuration(object_under_test):
```
Comment (Emar-Kar) [inline bigquery/tests/unit/model/test_model.py]: I think this should be renamed to `test_encryption_configuration_setter` for the code unification across [files](https://github.com/googleapis/google-cloud-python/blob/7c9c0cbd5c83c40e40fd86ea04907b7fe3c9c238/bigquery/tests/unit/test_table.py#L1120).

File: bigquery/tests/unit/model/test_model.py [hunk 2]
```diff
@@ -283,6 +299,18 @@ def test_replace_labels(object_under_test):
...
+    from google.cloud.bigquery.table import EncryptionConfiguration
+
+    assert not object_under_test.encryption_configuration
+    object_under_test.encryption_configuration = EncryptionConfiguration(
+        kms_key_name=KMS_KEY_NAME
+    )
+    assert object_under_test.encryption_configuration.kms_key_name == KMS_KEY_NAME
+    object_under_test.encryption_configuration = None
+    assert not object_under_test.friendly_name
```
Comment (Emar-Kar) [inline bigquery/tests/unit/model/test_model.py]: Do you mean `self.assertIsNone(object_under_test.encryption_configuration)`?

File: bigquery/google/cloud/bigquery/model.py [hunk 3]
```diff
@@ -256,6 +258,30 @@ def labels(self, value):
             value = {}
         self._properties["labels"] = value
 
+    @property
+    def encryption_configuration(self):
+        """google.cloud.bigquery.table.EncryptionConfiguration: Custom
+        encryption configuration for the table.
```
Comment (Emar-Kar) [inline bigquery/google/cloud/bigquery/model.py]: I assume it should be called **the model**.

File: bigquery/tests/unit/model/test_model.py (hunk 1 above)
Comment (HemangChothani) [reply bigquery/tests/unit/model/test_model.py]: In that file they use that naming convention like `test_set_description`, `test_set_friendly_name` so for the unification i gave the same name.

File: bigquery/tests/unit/model/test_model.py (hunk 2 above)
Comment (HemangChothani) [reply bigquery/tests/unit/model/test_model.py]: In the file they use same pattern for asset and  it's not in the class so self does't mean.

Comment (Emar-Kar) [reply bigquery/tests/unit/model/test_model.py]: I misspelled it. I mean that this test is for the `encryption_configuration`. You assert the `friendly_name` instead.

Comment (HemangChothani) [reply bigquery/tests/unit/model/test_model.py]: yes,that is true i changed it , please verify it.

**Comment by HemangChothani** [comment]
Merged.

**Pull request closed without merging**